src/search.c

```diff
@@ -2297,3 +2337,59 @@ void search_find_again(gboolean change_direction)
 				toolbar_get_widget_child_by_name("SearchEntry"), (result > -1));
 	}
 }
+
+
+static void load_combo_box_history(GtkWidget *combo_entry, gchar **recent_terms)
+{
+	guint i, len = 0;
+
+	if (recent_terms != NULL)
+	{
+		len = g_strv_length(recent_terms);
+		for (i = 0; (i < len); i++)
+		{
+			ui_combo_box_add_to_history(GTK_COMBO_BOX_TEXT(combo_entry), recent_terms[i], 0);
+		}
+	}
+}
+
+
+static void save_combo_box_history(GtkComboBox *combo, gchar ***recent_terms)
+{
+	guint i = 0;
+	gchar *combo_text;
+	GtkTreeIter iter;
+	GtkTreeModel *model;
+	gsize count;
+
+	model = gtk_combo_box_get_model(combo);
+	count = gtk_tree_model_iter_n_children(model, NULL);
+
+	g_strfreev(*recent_terms);
+
+	if (count == 0)
+	{
+		*recent_terms = NULL;
+		return;
+	}
+
+	*recent_terms = g_new0(gchar*, count + 1);
+
+	if (gtk_tree_model_get_iter_first(model, &iter))
+	{
+		do
+		{
+			gtk_tree_model_get(model, &iter, 0, &combo_text, -1);
+			if (*combo_text != 0)
+			{
+				(*recent_terms)[i] = g_strdup(combo_text);
```

Comment:
you could avoid having to duplicate the `combo_text` if you freed it conditionally:

```C
if (NZV(combo_text))
{
	(*recent_terms)[i] = combo_text;
}
else
{
	g_free(combo_text);
}
```